ledger/ledger.go

```diff
@@ -584,6 +584,14 @@ func (l *Ledger) trackerEvalVerified(blk bookkeeping.Block, accUpdatesLedger led
 	return eval(context.Background(), accUpdatesLedger, blk, false, nil, nil)
 }
 
+// IsWritingCatchpointFile returns true when a catchpoint file is being generated. The function is used by the catchup service so
+// that the memory pressure could be decreased until the catchpoint file writing is complete.
+func (l *Ledger) IsWritingCatchpointFile() bool {
+	l.trackerMu.RLock()
```

Comment:
why are the locks necessary?

Reply:
you're right. It was my bad. It should have been a `l.trackerMu.Lock()`.
explain:
`catchpointWriting` is being updated in two locations:
1. `initializeFromDisk`
2. `committedUpTo`

Since the channel itself is changing, it has to be protected. The former isn't an issue, but the latter is.
Given that the `committedUpTo` is updating the `catchpointWriting` while having the `l.trackerMu.RLock`, we want to have the `IsWritingCatchpointFile` taking the *Write* lock.

Reply:
It seems a little counterintuitive to me to have a getter called `IsWritingCatchpointFile` take a write lock...

Reply:
(especially when `committedUpTo` is the actual writer of the protected state)

Reply:
This does feel strange. I think it's an artifact of committedUpTo writing but taking only a read lock

Reply:
I personally feel it would be easier in the long run to read the code if `au.catchpointWriting` were replaced by something from `sync/atomic`.

Reply:
I think that the reason it appear weird is because it's a different "synchronization set". It should either be a it's own mutex + bool, or an atomic variable. I lean toward the latter. Either way, I think that we should avoid doing that for this release, and make it toward the next one.